components/toast_wrapper/toast_wrapper.jsx

```diff
@@ -0,0 +1,253 @@
+        }
+
+        return {
+            unreadCount,
+            showUnreadToast,
+            showNewMessagesToast,
+            channelMarkedAsUnread: props.channelMarkedAsUnread,
+            toastTimer,
+        };
+    }
+
+    componentDidMount() {
+        this.mounted = true;
+        this.forceShowUnreadToastTimer = setTimeout(() => {
+            this.forceUnreadEvenAtBottom = false;
+        }, TOAST_FADEOUT_TIME_UNREAD);
+        document.addEventListener('keydown', this.handleShortcut);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.atBottom && this.props.atBottom && this.props.atLatestPost) {
+            if (this.state.showNewMessagesToast) {
+                this.hideNewMessagesToast(false);
+            }
+
+            if (this.state.showUnreadToast) {
+                this.hideUnreadToast(false);
+            }
+        }
+
+        const prevPostsCount = prevProps.postListIds.length;
+        const presentPostsCount = this.props.postListIds.length;
+        const postsAddedAtBottom = presentPostsCount !== prevPostsCount && this.props.postListIds[0] !== prevProps.postListIds[0];
+        const notBottomWithLatestPosts = !this.props.atBottom && this.props.atLatestPost && presentPostsCount > 0;
+
+        //Marking existing messages as read based on last time user reached to the bottom
+        //This moves the new message indicator to the latest posts and keeping in sync with the toast count
+        if (postsAddedAtBottom && notBottomWithLatestPosts && !this.state.showUnreadToast) {
+            this.props.updateNewMessagesAtInChannel(this.props.lastViewedBottom);
+        }
+    }
+
+    componentWillUnmount() {
+        clearTimeout(this.forceShowUnreadToastTimer);
+        document.removeEventListener('keydown', this.handleShortcut);
+    }
+
+    handleShortcut = (e) => {
+        if (Utils.isKeyPressed(e, Constants.KeyCodes.ESCAPE)) {
+            if (this.state.showUnreadToast) {
+                this.hideUnreadToast();
+            } else if (this.state.showNewMessagesToast) {
+                this.hideNewMessagesToast();
+            }
+        }
+    };
+
+    hideUnreadToast = (dismiss = true) => {
+        if (this.state.showUnreadToast) {
+            let toastTimer = this.forceUnreadEvenAtBottom ? TOAST_FADEOUT_TIME_UNREAD : TOAST_FADEOUT_TIME;
+            if (dismiss) {
+                toastTimer = 0;
+            }
+            this.setState({
+                showUnreadToast: false,
+                toastTimer,
+            });
+        }
+    }
+
+    hideNewMessagesToast = (dismiss = true) => {
+        if (this.state.showNewMessagesToast) {
+            this.setState({
+                showNewMessagesToast: false,
+                toastTimer: dismiss ? 0 : TOAST_FADEOUT_TIME,
+            });
+            if (dismiss) {
+                this.props.updateLastViewedBottomAt();
+            }
+        }
+    }
+
+    newMessagesToastText = (count, since) => {
```

Comment:
This still needs to be fixed. We should just have two messages and alternate between since combining strings doesn't work the same in all languages.